csharp/source/autowrap/csharp/csharp.d

```diff
@@ -0,0 +1,1104 @@
+module autowrap.csharp.csharp;
+
+import autowrap.csharp.common : LibraryName, RootNamespace, camelToPascalCase, getDLangSliceInterfaceName, getDLangInterfaceName;
+import autowrap.reflection : AllFunctions, AllAggregates, isModule;
+
+import std.ascii : newline;
+import std.meta: allSatisfy, AliasSeq;
+import std.traits : isArray, fullyQualifiedName, moduleName, isFunction, Fields, FieldNameTuple, hasMember, functionAttributes, FunctionAttribute, ReturnType, Parameters, ParameterIdentifierTuple;
+import std.string : format, replace;
+import std.conv : to;
+
+private string[string] returnTypes;
+private CSharpNamespace[string] namespaces;
+private csharpRange rangeDef;
+
+enum string voidTypeString = "void";
```

Comment:
I'm not sure these are better than the string literals.

Reply:
And in the previously in this review you said you didn't want the string literals. Please make up your mind then stick with it.